### ISSUE TITLE
Adding Categories?

### ISSUE THREAD

**Issue by FreeApophis** [amis92/csharp-source-generators#165]
Love the list, Source Generators are really neat.

But, what about creating categories?

There are about a dozen or so Dependency Injection Framworks in the list, and it would be really nice to have them in one place.

I could make a PR for that if there is interest in the idea.

**Comment by amis92** [comment]
Yeah, sounds like a good idea - I'll gladly accept such a contribution.

What categories do you have in mind?

- Dependency Injection is definitely a recurring one
- Enums, discriminated unions, value objects - all seem to be a similar area
- Mappers
- Functional - a lot of sgens attempt to simulate or make easier-to-use functional patterns like Options, Results etc
- Serialization
- Testing/Tests - some generators dedicated to testing, mocking etc
- ???

**Comment by FreeApophis** [comment]
I tried my best: I had to look into a lot of ReadMe's to better understand the intention of each project, but I think it was worth the effort.

I hope you find the categories clear and useful, it was sometimes a bit tricky, some projects could be in different categories.

Some things to note:

* The TOC is done by hand. But I think it is a good thing to have an overview, but it might be too difficult to maintain.
* The order of the catergories is not really well defined, it just grew like that.
* Each categorie is alphabetically sorted again
